lib/User_LDAP.php

```diff
@@ -182,11 +182,18 @@ public function getUsers($search = '', $limit = 10, $offset = 0) {
 			$filter,
 			$this->access->userManager->getAttributes(true),
 			$limit, $offset);
-		$ldap_users = $this->access->ownCloudUserNames($ldap_users);
-		\OCP\Util::writeLog('user_ldap', 'getUsers: '.count($ldap_users). ' Users found', \OCP\Util::DEBUG);
+		$ldap_users2 = $this->access->ownCloudUserNames($ldap_users);
+		\OCP\Util::writeLog('user_ldap', 'getUsers: '.count($ldap_users2). ' Users found', \OCP\Util::DEBUG);
+
+		// sanity check
+		$ldap_users_count = count($ldap_users);
+		$ldap_users2_count = count($ldap_users2);
+		if ($ldap_users_count !== $ldap_users2_count) {
+			\OCP\Util::writeLog('user_ldap', "getUsers: fetched $ldap_users_count users but returned $ldap_users2_count", \OCP\Util::ERROR);
+		}
 
-		$this->access->connection->writeToCache($cachekey, $ldap_users);
-		return $ldap_users;
+		$this->access->connection->writeToCache($cachekey, $ldap_users2);
+		return $ldap_users2;
```

Comment:
I guess that if this method gets 500 users and only 498 of them are "good" then it could query again for the next users, and put the first 2 good ones into the array to return to the caller.
But then when it is called in future with some offset, it needs to know how to adjust that to also skip the "non-good" users that were in previous batches.
And of course this whole thing can be a problem if users are being added or deleted on the LDAP server between query batches - then where the oC end thinks it is up to will be out-of-alignment with LDAPs current set of users.